packages/mobile/src/components/audio/GoogleCast.tsx

```diff
@@ -130,24 +142,51 @@ export const useChromecast = () => {
     }
   }, [playing, playCast, pauseCast, castState])
 
+  // Set buffering state when cast is buffering
+  useEffect(() => {
+    if (
+      castState === CastState.CONNECTING ||
+      ((mediaStatus === undefined ||
+        mediaStatus?.playerState === undefined ||
+        mediaStatus?.playerState === MediaPlayerState.IDLE ||
+        mediaStatus?.playerState === MediaPlayerState.LOADING ||
+        mediaStatus?.playerState === MediaPlayerState.BUFFERING) &&
+        castState !== CastState.NOT_CONNECTED)
+    ) {
+      dispatch(playerActions.setBuffering({ buffering: true }))
+    } else {
+      dispatch(playerActions.setBuffering({ buffering: false }))
+    }
+  }, [mediaStatus, castState, dispatch])
+
   // Seek the cast device
   useEffect(() => {
     if (seek !== null) {
       client?.seek({ position: seek })
     }
   }, [client, seek])
 
+  // Mute the track player if we are connecting to cast
   useEffect(() => {
     if (
       castState === CastState.CONNECTED ||
       castState === CastState.CONNECTING
     ) {
       TrackPlayer.setVolume(0)
-    } else {
-      TrackPlayer.setVolume(1)
     }
   }, [castState])
 
+  // Handle disconnection from cast device
+  useEffect(() => {
+    if (
+      castState === CastState.NOT_CONNECTED &&
+      previousCastState === CastState.CONNECTED
+    ) {
+      TrackPlayer.setVolume(1)
+      dispatch(playerActions.pause())
```

Comment:
nice so we want to pause when disconnected rather than keep playing? prob makes sense, like when headphones disconnect?

Reply:
I'm not actually sure, but it was not behaving correctly the other way and wanted to get something out and I think that this is "acceptable" behavior for now. Interestingly `TrackPlayer.play()` does nothing here unless given a >1s timeout. Not sure what it is, but figured getting a fix in first would be good.